packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -243,3 +234,68 @@ export const getCardStyles = stylesFactory((theme: GrafanaTheme, disabled = fals
     `,
   };
 });
+
+interface ChildProps {
+  styles?: ReturnType<typeof getCardStyles>;
+  disabled?: boolean;
+}
+
+const Tags: FC = ({ children }) => {
+  return <>{children}</>;
+};
+Tags.displayName = 'Tags';
+
+const Figure: FC<ChildProps> = ({ children, styles }) => {
+  return <div className={styles?.media}>{children}</div>;
+};
+
+Figure.displayName = 'Figure';
+
+const Meta: FC<ChildProps> = ({ children, styles }) => {
+  // Join meta data elements by '|'
+  const meta = useMemo(
+    () =>
+      Array.isArray(children)
+        ? React.Children.toArray(children)
+            .filter(Boolean)
+            .reduce((prev, curr, i) => [
+              prev,
+              <span key={`separator_${i}`} className={styles?.separator}>
+                |
+              </span>,
+              curr,
+            ])
+        : children,
+    [children, styles?.separator]
```

Comment:
the readability of this is low for me (using ? operator that stretches many lines), wrapped in inline function that stretches many lines (without curly braces),  Why the filter(Boolean) ? why the reduce? 

The memoization also seems unnecessary and could be put on the component instead to make it a pure component.

Reply:
Seems like filter isn't necessary indeed. Reduce is needed to join multiple metadata elements by a horizontal line. Agree that memoization could be moved onto component itself.